Services/RemoteShopService.php

```diff
@@ -0,0 +1,125 @@
+<?php
+/**
+ * (c) shopware AG <info@shopware.com>
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ShopwarePlugins\Connect\Services;
+
+use SebastianBergmann\GlobalState\RuntimeException;
+use Shopware\Connect\SDK;
+use ShopwarePlugins\Connect\Components\ConfigFactory;
+use ShopwarePlugins\Connect\Components\SnHttpClient;
+
+/**
+ * Class RemoteShopService
+ * @package ShopwarePlugins\Connect\Services
+ */
+class RemoteShopService
+{
+    /**
+     * @var SDK
+     */
+    private $sdk;
+
+    /**
+     * RemoteShopService constructor.
+     * @param SDK $sdk
+     */
+    public function __construct(SDK $sdk)
+    {
+        $this->sdk = $sdk;
+    }
+
+    /**
+     * @param $shopId
+     * @return bool
+     */
+    public function isPingRemoteShopSuccessful($shopId)
+    {
+        try {
+            return $this->tryPingRemoteShop($shopId);
+        } catch (\RuntimeException $exception) {
+            return $this->handlePingRemoteShopError($shopId, $exception);
+        }
+    }
+
+    /**
+     * @param $shopId
+     * @return bool
+     */
+    private function tryPingRemoteShop($shopId)
+    {
+        if ($this->sdk->pingShop($shopId) !== 'pong') {
+            throw new RuntimeException("Ping was not returning excepted result 'pong'.");
+        } else {
+            return true;
```

Comment:
is the else case obsolete?

Reply:
I had the idea to let `isPingRemoteShopSuccessful` return either true or false. Because I only check if it is false inside Article.php it would also work without return true. But it felt a bit weird to me.